pkg/hs/king/lib/Ur/Vere/Pier.hs

```diff
@@ -130,18 +130,36 @@ booted pill lite flags ship boot = do
 -- Resume an existing ship. ----------------------------------------------------
 
 resumed :: (HasPierConfig e, HasLogFunc e)
-        => Serf.Flags
+        => Maybe Word64 -> Serf.Flags
         -> RAcquire e (Serf e, EventLog, SerfState)
-resumed flags = do
+resumed event flags = do
     top    <- view pierPathL
+    tap    <- case event of
+                Nothing -> pure top
+                Just ev -> rio $ do
+                  snap <- getSnapshot top ev
+                  case snap of
+                    Nothing -> pure top
+                    Just sn -> pure sn
+    rio $ logTrace $ displayShow tap
     log    <- Log.existing (top <> "/.urb/log")
-    serf   <- Serf.run (Serf.Config top flags)
-    serfSt <- rio $ Serf.replay serf log
+    serf   <- Serf.run (Serf.Config tap flags)
+    serfSt <- rio $ Serf.replay serf log event
 
     rio $ Serf.snapshot serf serfSt
 
     pure (serf, log, serfSt)
 
+getSnapshot :: String -> Word64 -> RIO e (Maybe String)
+getSnapshot top last = do
+  createDirectoryIfMissing True $ top <> "/.partial-replay/"
+  snapshots <- listDirectory $ top <> "/.partial-replay/"
+  let nums = Ur.Prelude.mapMaybe readMay snapshots :: [Int]
+  let snaps = sort $ Ur.Prelude.filter (\n -> n <= (fromIntegral last)) nums
+  case (Data.List.uncons $ reverse snaps) of
+    Nothing      -> pure Nothing
+    Just (sn, _) -> pure $ Just $ top <> "/.partial-replay/" <> (show sn)
+
```

Comment:
I would write it like this:

```
getSnapshot :: forall e. FilePath -> Word64 -> RIO e (Maybe FilePath)
getSnapshot top last = do
    lastSnapshot <- lastMay <$> listReplays
    pure (replayToPath <$> lastSnapshot)
  where
    replayDir        = top </> ".partial-replay"
    replayToPath eId = replayDir </> show eId

    listReplays ∷ RIO e [Word64]
    listReplays = do
        createDirectoryIfMissing True replayDir
        snapshotNums <- mapMaybe readMay <$> listDirectory replayDir
        pure $ sort (filter (<= fromIntegral last) snapshotNums)
```

Reply:
I don't quite understand what this does by looking at the code. Could this be named `lastPartialReplaySnapshot`?